enhancements/network/communication-ingress-flows-matrix.md

```diff
@@ -0,0 +1,198 @@
+  OpenShift communication ingress flows matrix.
+
+- Keep the communication matrix documented in OpenShift release documents
+  updated and validate it does not change.
+
+### Non-Goals
+
+- Egress traffic.
+
+## Proposal
+
+We introduce a new module to generate an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- A communication matrix describing the expected flows of incoming traffic will 
+  be included in every OpenShift release documentation.
+
+- A new `oc` command will be added to generate a current snapshot of known 
+  listening ports in a running cluster, `oc adm generate-commatrix`. #Sabina: what should be the command name?
+
+- A new option will be added to OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### Workflow Description
+
+An OpenShift administrator would like to get an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- The admin reviews OpenShift release documentation to get the included communication
+  matrix describing the expected flows of incoming traffic.
+
+- The admin uses the OpenShift command-line interface (CLI) to generate an up-to-date 
+  communication matrix using the following command `oc adm generate-commatrix`.
+
+- Optionally, the admin uses the OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints
+
+A new helper module `node-comm-lib` will be added to `openshift/library-go`. #Sabina: what should be the module name?
+The module will be used to create a communication ingress flows matrix 
+from the existing `EndpointSlices` in the OpenShift cluster.
+
+`EndpointSlices` group network endpoints together by unique combinations of 
+protocol, port number, and `Service` name and therefore can be used as the 
+source of truth for cluster traffic. For that to work, every application that 
+attracts ingress traffic running in the cluster should have a `Service` object,
+the control plane automatically creates `EndpointSlices` for any `Service` that 
+has a selector specified.
+
+To generate the communication matrix the `EndpointSlices` in the cluster should 
+be filtered by `Service` type `NodePort` and `LoadBalancer` as there are also 
+`EndpointSlices` in the cluster for internal `Services`. The node hosted services 
+such as sshd, rpc, etc. are currently missing `EndpointSlices` and should be created 
+by the Machine API Operator during cluster deployment.
+
+Sabina to Yuval: What k8s link you wanted to add here?
+
+The module allows to generate the communication matrix in various formats:
+- YAML/JSON
+- CSV
+- nftables
+
+Each record describes a flow with the following information:
+```
+traffic-direction     Data flow direction (ingress/egress)
+protocol              Network protocol (TCP/UDP)
+port-number           Flow port number
+service-name          OpenShift Service name
```

Comment:
do we want to include more info such as container/pod name and maybe even the process name?

I'm thinking in the lines of how to make this easy for anyone who externally runs nmap to correlated with this.
so the protocol+port is the unique identifier, but it'll be weird (for them) if nmap shows "kube-rbac-proxy" and we say "MCO", so maybe we want to show that it's the kube-rbac-proxy that is inside MCO (just kube-rbac-proxy is also bad, because you cannot understand to what deployment it belongs)